lib/amq/uri.rb

```diff
@@ -26,8 +37,28 @@ def self.parse(connection_string)
         opts[:vhost] = ::CGI::unescape($1)
       end
 
+      if uri.query
+        query_params = CGI::parse(uri.query)
+
+        normalized_query_params = Hash[query_params.map { |param, value| [param, value.one? ? value.first : value] }]
+
+        opts[:heartbeat] = normalized_query_params["heartbeat"].to_i
+        opts[:connection_timeout] = normalized_query_params["connection_timeout"].to_i
+        opts[:channel_max] = normalized_query_params["channel_max"].to_i
+        opts[:auth_mechanism] = normalized_query_params["auth_mechanism"]
+
+        %w(verify fail_if_no_peer_cert cacertfile certfile keyfile).each do |ssl_option|
+          if normalized_query_params[ssl_option] && uri.scheme == "amqp"
+            raise ArgumentError.new("Only of use for the amqps scheme")
```

Comment:
This message doesn't clarify what is "of use" (can be used). Please add it.

Reply:
Sure what does sound the best?
1. `"'varify' TLS option is applicable only for 'amqps' schema"`
2. `"'varify' option should be used only within 'amqps' schema"`
3. `TLS options are relevant only for 'amqps' schema"`

Reply:
I don't mind to convert it to the warning, but an exception could attract attention to the possible misspelling of the schema. Maybe you already have a convention in this or other `ruby-amqp` libraries, so it will be natural to keep consistent behavior across all of them.

Reply:
OMG, it was so fast. I feel like talk too much ^_^ But I'd like to make a follow-up PR with changes we agreed here, particularly rename the message.
Leave the exception or change to the warning?

Reply:
I did change the message in https://github.com/ruby-amqp/amq-protocol/commit/50dbf6b70dca6906de01a34fe13df4c9f6177bae but because I cannot push that to your repo, it won't show up here.

Reply:
Yeah, I noticed when fetched upstream ^_^ That's OK. BTW, is it preferable to contribute from forked repo or straight to the initial repo?

Reply:
You don't have the permissions to push to the mainline. I see no reason to not pull from and branch off of it. I don't really care as long as the PR is against the correct branch and is something I'd accept.